access-token-management/src/AccessTokenManagement.OpenIdConnect/UserTokenEndpointService.cs

```diff
@@ -49,17 +26,24 @@ public async Task<UserToken> RefreshAccessTokenAsync(
     {
         var refreshToken = userToken.RefreshToken ?? throw new ArgumentNullException(nameof(userToken.RefreshToken));
 
-        _logger.LogTrace("Refreshing refresh token: {token}", refreshToken);
 
-        var oidc = await _configurationService.GetOpenIdConnectConfigurationAsync(parameters.ChallengeScheme).ConfigureAwait(false);
+        var oidc = await configurationService.GetOpenIdConnectConfigurationAsync(parameters.ChallengeScheme).ConfigureAwait(false);
+
+        // Add the ClientID to all subsequent log messages
+        using var logScope = logger.BeginScope(
+            (LogMessages.Parameters.ClientId, oidc.ClientId)
+        );
+
+        logger.RefreshingAccessTokenUsingRefreshToken(refreshToken, hashAlgorithm: Crypto.HashData);
```

Comment:
Before, the refresh tokens where logged in debug logging. Now, we decided that the refresh tokens shouldn't be in the logs, but their hash is OK.